pkg/virt-controller/watch/vm.go

```diff
@@ -2356,3 +2350,39 @@ func (c *VMController) resolveControllerRef(namespace string, controllerRef *v1.
 	}
 	return vm.(*virtv1.VirtualMachine)
 }
+
+func (c *VMController) storeFlavorControllerRevisions(vm *virtv1.VirtualMachine) (*flavorv1alpha1.VirtualMachineFlavorSpec, *flavorv1alpha1.VirtualMachinePreferenceSpec, error) {
+
+	logger := log.Log.Object(vm)
+	flavorRevision, flavorSpec, err := c.flavorMethods.StoreFlavorRevision(vm)
+	if err != nil {
+		logger.Reason(err).Error("Failed to store ControllerRevision of VirtualMachineFlavorSpec for the Virtualmachine.")
+		return nil, nil, err
+	}
+
+	preferenceRevision, preferenceSpec, err := c.flavorMethods.StorePreferenceRevision(vm)
+	if err != nil {
+		logger.Reason(err).Error("Failed to store ControllerRevision of VirtualMachinePreferenceSpec for the Virtualmachine.")
+		return nil, nil, err
+	}
+
+	if flavorRevision != nil || preferenceRevision != nil {
+
+		//TODO - Should we fetch the VirtualMachine here one final time to ensure we are up to date?
+		if flavorRevision != nil {
+			vm.Spec.Flavor.RevisionName = flavorRevision.Name
+		}
+
+		if preferenceRevision != nil {
+			vm.Spec.Preference.RevisionName = preferenceRevision.Name
+		}
+
+		_, err = c.clientset.VirtualMachine(vm.Namespace).Update(vm)
```

Comment:
I'd prefer to see a Patch here instead of an Update.

We have a few other Update() calls within the sync logic which are kind of problematic. What happens is, if two bits of logic attempt to perform an Update() on a vm object in the same reconcile loop, only the first one passes and the subsequent ones fail due to using a stale VM. If all the places that perform an "Update" use Patch instead, they won't collide unless two Patches attempt to mutate the same field

Reply:
ACK thanks for the context and yeah fully agree. I've moved over to using patch.